src/main/java/de/refactoringbot/services/wit/WitService.java

```diff
@@ -0,0 +1,170 @@
+		try {
+			// Create object
+			BotIssue issue = new BotIssue();
+
+			// Add data to comment
+			issue.setCommentServiceID(comment.getCommentID().toString());
+			issue.setLine(comment.getPosition());
+			issue.setFilePath(comment.getFilepath());
+
+			// Set all Java-Files and Java-Roots
+			List<String> allJavaFiles = fileService.getAllJavaFiles(gitConfig.getRepoFolder());
+			issue.setAllJavaFiles(allJavaFiles);
+			issue.setJavaRoots(fileService.findJavaRoots(allJavaFiles));
+
+			mapCommentBodyToIssue(issue, comment.getCommentBody());
+			return issue;
+		} catch (WitAPIException | CommentUnderstandingMessage e) {
+			logger.error(e.getMessage(), e);
+			throw new CommentUnderstandingMessage(e.getMessage());
+		} catch (IOException e) {
+			logger.error(e.getMessage(), e);
+			throw new IOException("Could not create a BotIssue from the comment '" + comment.getCommentBody() + "'!");
+		}
+	}
+
+	/**
+	 * This method maps the information of a comment to the issue.
+	 * 
+	 * @param issue
+	 * @param commentBody
+	 * @throws WitAPIException
+	 * @throws CommentUnderstandingMessage
+	 */
+	private void mapCommentBodyToIssue(BotIssue issue, String commentBody)
+			throws WitAPIException, CommentUnderstandingMessage {
+		// Anonymize possible sensitive data
+		commentBody = dataAnonymizer.anonymizeComment(commentBody);
+		// Log anonymized message as info
+		logger.info("Anonymized comment: " + commentBody);
+		// Get Wit-Object from Wit-API
+		WitObject witObject = witDataGrabber.getWitObjectFromComment(commentBody);
+
+		// If wit returns multiple operations or none refactorings
+		if (witObject.getEntities().getRefactoring().size() != 1) {
+			throw new CommentUnderstandingMessage("Not sure what refactoring to execute!");
+		}
+
+		// Read unique operation + object
+		WitEntity refactoring = witObject.getEntities().getRefactoring().get(0);
+
+		// If Add-Annotation refactoring is sure
+		if (refactoring.getValue().equals("ADD-ANNOTATION")) {
+			// If 0/2+ possilbe annotations returned
+			if (witObject.getEntities().getJavaAnnotations().size() != 1) {
+				throw new CommentUnderstandingMessage("Not sure which annotation you want to add!");
+			}
+			// Read unique java annotation
+			WitEntity javaAnnot = witObject.getEntities().getJavaAnnotations().get(0);
+			// If override annotation returned
+			if (javaAnnot.getValue().equals("Override")) {
+				issue.setRefactoringOperation(RefactoringOperations.ADD_OVERRIDE_ANNOTATION);
+				// If some unknown annotation returned
+			} else {
+				throw new CommentUnderstandingMessage("Adding '" + javaAnnot.getValue() + "' is not supported!");
+			}
+		}
+
+		// If Reorder-Modifier
+		else if (refactoring.getValue().equals("REORDER-MODIFIER")) {
+			// Annotations or refactoring strings are not involved
+			if (witObject.getEntities().getRefactoringString().size() != 0
+					|| witObject.getEntities().getJavaAnnotations().size() != 0) {
+				logger.warn(
```

Comment:
Is this for the case that, for example, a user requested more than one refactoring within the same comment?

Reply:
The REORDER-MODIFIER does not use/contain a refactoringString or javaAnnotation. If wit returns a refactoringString or javaAnnotation with a REORDER-MODIFIER refactoring, then we throw a warning but not an error (since the reorder modifier class does not use those 2 and it does not matter if we get them from wit or not -> refactoring should proceed correctly). This logged warning can be used so we can find and fix those expressions easier in our wit-inbox.